### PR TITLE
[ADD] sale_global_discount: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Apply global financial discounts to sales that will be transmited to invoices
and accounting.

@Tecnativa TT20971

### CODE REVIEW

**Comment by rafaelbn** [comment]
Hello @fclementic2c @legalsylvain , about this: https://twitter.com/Frederic_C2C/status/1245984823925186560

With this module you can set a different financial account for the global discount.

**Review by Tardo — APPROVED** [review]
:+1: LGTM, code review.

File: sale_global_discount/README.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. Go to *Settings > Parameters > Global Discounts*
+#. Add a new discount that can be either by percentage or fixed amount.
```
Comment (pedrobaeza) [inline sale_global_discount/README.rst]: fixed amount?

Comment (chienandalu) [reply sale_global_discount/README.rst]: No, that of course isn't right. Fixed.

File: sale_global_discount/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    )
+    amount_total_before_global_discounts = fields.Monetary(
+        string='Amount Untaxed Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+
+    @api.model
+    def get_discounted_global(self, price=0, discounts=None):
```
Comment (manuelcalerosolis) [inline sale_global_discount/models/sale_order.py:36]: @api.model
    def get_discounted_global(self, price=0, discounts=[]):
        """Compute discounts successively"""
        if not discounts:

**Review by manuelcalerosolis — APPROVED** [review]
Congratulations, great work and very useful.

File: sale_global_discount/models/sale_order.py (hunk 2 above)
Comment (chienandalu) [reply sale_global_discount/models/sale_order.py:36]: Sorry, what do you mean? :sweat_smile:

**Comment by tonygalmiche** [comment]
please @pedrobaeza could you have a look ?
is a review necessary?

File: sale_global_discount/models/sale_order.py (hunk 2 above)
Comment (rousseldenis) [reply sale_global_discount/models/sale_order.py:36]: He wants you to change the signature function replacing None by [].

But @manuelcalerosolis, this is not good as explained : https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

**Review by rousseldenis — APPROVED** [review]
Code review

File: sale_global_discount/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            })
+        return res
+
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        res = super().onchange_partner_id()
+        if self.partner_id.customer_global_discount_ids:
+            self.global_discount_ids = (
+                self.partner_id.customer_global_discount_ids or
+                self.parnter_id.commercial_partner_id
```
Comment (manuelcalerosolis) [inline sale_global_discount/models/sale_order.py:85]: @chienandalu correct this typo error "parnter_id"

Comment (chienandalu) [reply sale_global_discount/models/sale_order.py:85]: Thanks @manuelcalerosolis Can you do a quick PR?

**Pull request merged**